.github/workflows/rust_checks.yml

```diff
@@ -0,0 +1,127 @@
+name: Rust Static checks
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/policy_controller.yml
+      - Cargo.lock
+      - Cargo.toml
+      - charts/linkerd-control-plane/templates/destination-rbac.yaml
+      - charts/linkerd-crds/templates/policy/**
+      - deny.toml
+      - policy-controller/**
+      - policy-test/**
+      - rust-toolchain
+
+permissions:
+  contents: read
+
+env:
+  CARGO_INCREMENTAL: 0
+  CARGO_NET_RETRY: 10
+  K3D_VERSION: v5.4.1
+  PROTOC_NO_VENDOR: 1
+  PROXY_INIT_VERSION: v1.5.3
+  RUST_BACKTRACE: short
+  RUSTUP_MAX_RETRIES: 10
+
+
+jobs:
+  fmt:
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    container:
+      image: docker://rust:1.60.0
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+      - run: rustup component add rustfmt
+      - run: cargo fmt --all -- --check
```

Comment:
can we use @olix0r's `cargo-action-fmt` for displaying rustfmt output as well?

Reply:
No, we can't.

```
:; cargo fmt --all -- --check            
Diff in /workspaces/k8s-gateway-api-rs/src/lib.rs at line 9:
 // CustomResource derivations.
 
 /// The v1alpha2 API version.
-pub mod   v1alpha2;
+pub mod v1alpha2;
```

cargo-action-fmt needs JSON-formatted output